app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -1,15 +1,97 @@
 package com.lightricks.feedexercise.data
 
+import android.content.Context
+import android.util.Log
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
-import androidx.test.ext.junit.runners.AndroidJUnit4
-import org.junit.runner.RunWith
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import com.google.common.truth.Truth.assertThat
+import com.lightricks.feedexercise.database.FeedDao
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedItemEntity
+import com.lightricks.feedexercise.network.MockFeedApiService
+import com.lightricks.feedexercise.util.Event
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
-@RunWith(AndroidJUnit4::class)
-class FeedRepositoryTest {
-   //todo: add the tests here
+class SimpleEntityReadWriteTest2 {
+    private lateinit var feedDao: FeedDao
+    private lateinit var db: FeedDatabase
+    private lateinit var service: MockFeedApiService
+
+    @get:Rule
+    val executorRule: InstantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @Before
+    fun createDb() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        db = Room.inMemoryDatabaseBuilder(
+            context, FeedDatabase::class.java
+        ).build()
+        feedDao = db.feedDao()
+    }
+    @Before
+    fun createService(){
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        service = MockFeedApiService(context)
+    }
+
+    @After
+    fun closeDb() {
+        db.close()
+    }
+
+    @Test
+    fun refreshSavesDataAndFeedItemsContainsData(){
```

Comment:
This test could be split into 2 tests:
1. Refresh the repository and verify that the items were written to the DB (by reading directly from `FeedDao`)
2. Insert something into `FeedDao` and verify that `getFeedItems` reflects the items.